security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -433,17 +433,18 @@ func (sc *SecretManagerClient) generateKeyCertFromExistingFiles(certChainPath, k
 // readFileWithTimeout reads the given file with timeout. It returns error
 // if it is not able to read file after timeout.
 func (sc *SecretManagerClient) readFileWithTimeout(path string) ([]byte, error) {
-	retryBackoffInMS := int64(firstRetryBackOffInMilliSec)
+	retryBackoffInMS := time.Duration(firstRetryBackOffInMilliSec)
 	for {
 		cert, err := ioutil.ReadFile(path)
 		if err == nil {
 			return cert, nil
 		}
+		if retryBackoffInMS > totalTimeout {
```

Comment:
`totalTimeout` is the entire time. This checks the interval compared to the total time, which is not correct.

For example, if total time is 4 and interval starts at 1, we will only meet this criteria after 7 (1+2+4) seconds have passed?

Reply:
Thanks, updated